packages/devtools/lib/src/ui/custom.dart

```diff
@@ -151,25 +224,23 @@ class SelectableTree<T> extends CoreElement {
   void setItems(List<T> items) {
     this.items = items;
 
-    final bool hadSelection = _selectedElement != null;
-    _selectedElement = null;
+    final bool hadSelection = _selectedItem != null;
+    _selectedItem = null;
 
     clear();
 
-    for (T item in items) {
-      _populateInto(this, item);
-    }
+    treeItems = _populateItems(items, this, null);
 
-    if (hadSelection && _selectedElement == null) {
+    if (hadSelection && _selectedItem == null) {
       _selectionController.add(null);
     }
   }
 
-  void _populateInto(CoreElement parent, T item) {
+  TreeItem<T> _populateInto(CoreElement container, T item) {
```

Comment:
@devoncarew can you review these changes? I don't have a great feel of what sort of tree api you were going for.

Reply:
For some context - I added a TreeItem<T> to wrap the T's so that we have references to siblings/parents/children for each node (for the nodes that have been discovered), since we often have to traverse a fair bit of the tree for keyboard nav (for ex., when you press **Up**, we may move to the previous siblings last visible ancestor if the previous sibling is expanded).

It's possible we could tidy this up some, like having both `treeItems` and `items` on the tree feels a little strange. I'm happy to make changes, but wanted some feedback before I went too far :-)

Reply:
I was just chatting with Jacob about moving the navigation stuff to a mixin - I'd started doing it that way but then reverted it - but now I know there are two trees, I think it would make much more sense.

It's probably better to do that in another PR though, so it'd be good to get this one to a point we're happy to land first, then I'll start extracting stuff.